yt_dlp/extractor/motherless.py

```diff
@@ -246,3 +246,74 @@ def _get_page(idx):
             'description': description,
             'entries': playlist
         }
+
+
+class MotherlessGalleryIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?motherless\.com/GV?(?P<id>[A-F0-9]+)'
+    _TESTS = [{
+        'url': 'https://motherless.com/G338999F',
+        'info_dict': {
+            'id': '338999F',
+            'title': 'Random',
+        },
+        'playlist_mincount': 220,
+    }, {
+        'url': 'https://motherless.com/GVABD6213',
+        'info_dict': {
+            'id': 'ABD6213',
+            'title': 'Cuties',
+        },
+        'playlist_mincount': 1,
+    }, {
+        'url': 'https://motherless.com/GBCF7622',
+        'info_dict': {
+            'id': 'BCF7622',
+            'title': 'Vintage',
+        },
+        'playlist_mincount': 0,
+    }]
+
+    def _extract_entries(self, webpage, base):
+        entries = []
+        for mobj in re.finditer(r'href=".*(?P<href>\/[A-F0-9]+)"\s+title="(?P<title>[^"]+)', webpage):
+            video_url = compat_urlparse.urljoin(base, mobj.group("href"))
+            if not MotherlessIE.suitable(video_url):
+                continue
+
+            video_id = MotherlessIE._match_id(video_url)
+            title = mobj.group('title')
+            entries.append(self.url_result(
+                video_url, ie=MotherlessIE.ie_key(), video_id=video_id,
+                video_title=title))
+
+        return entries
+
+    def _real_extract(self, url):
+        gallery_id = self._match_id(url)
+        page_url = compat_urlparse.urljoin(url, f'/GV{gallery_id}')
+        webpage = self._download_webpage(page_url, gallery_id)
+        title = self._search_regex(
+            r'<title>([\w\s]+\w)\s+', webpage, 'title', fatal=False)
+        page_count = self._int(self._search_regex(
+            r'(\d+)</a><a[^>]+rel="next">',
+            webpage, 'page_count', default=1), 'page_count')
+        PAGE_SIZE = 60
+
+        def _get_page(idx):
+            if not page_count:
+                return
+            webpage = self._download_webpage(
+                page_url, gallery_id, query={'page': idx + 1},
+                note='Downloading page %d/%d' % (idx + 1, page_count)
+            )
+            for entry in self._extract_entries(webpage, url):
+                yield entry
```

Comment:
imo, this should just be part of `_extract_entries`

Reply:
Which bits exactly? This is the pager function that is being passed into `InAdvancedPagedList`.

Reply:
The whole thing. You can use `functools.partial` as needed

Reply:
Do you have an example extractor I can look at for that?

Reply:
ping me once other points are addressed, and I'll help with this

Reply:
I think most other points are.